### PR TITLE
Backend setup

### PR DESCRIPTION
**Pull request by PurpJell**

### CODE REVIEW

File: .gitignore
Comment (rimvydascivilis) [inline .gitignore:1]: Can we move this file to `/movieh`?

File: movieh/db.sqlite3
Comment (rimvydascivilis) [inline movieh/db.sqlite3:1]: Is this a necessary file if we're going to use PostgreSQL?

File: movieh/movieh/settings.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/5.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'django-insecure-vd6t$%08dvi5yhhzhxlq_!06ta9a5ecx-m%owpkym%x(6rcc=s'
```
Comment (rimvydascivilis) [inline movieh/movieh/settings.py]: Can we get secrets from env variables?

File: movieh/movieh/settings.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+WSGI_APPLICATION = 'movieh.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/5.1/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
```
Comment (rimvydascivilis) [inline movieh/movieh/settings.py]: Also get db config from env vars

File: movieh/movieh/urls.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    1. Import the include() function: from django.urls import include, path
+    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
+"""
+from django.contrib import admin
+from django.urls import path, include
+from django.contrib.auth.models import User
+from rest_framework import routers, serializers, viewsets
+
+# Serializers define the API representation.
+class UserSerializer(serializers.HyperlinkedModelSerializer):
```
Comment (rimvydascivilis) [inline movieh/movieh/urls.py]: `UserSerializer` class should be in `serializers.py` file

File: movieh/movieh/urls.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    class Meta:
+        model = User
+        fields = ['url', 'username', 'email', 'is_staff']
+
+# ViewSets define the view behavior.
+class UserViewSet(viewsets.ModelViewSet):
```
Comment (rimvydascivilis) [inline movieh/movieh/urls.py]: This class should be in `views.py` file

**Comment by rimvydascivilis** [comment]
Can we change `/movieh` to `/backend` for consistency, since we have `/frontend`?

**Review by rimvydascivilis — CHANGES_REQUESTED** [review]

**Comment by rimvydascivilis** [comment]
Could you add `requirements.txt` dependency file?

File: backend/movieh/urls.py [hunk 5]
```diff
@@ -16,19 +16,9 @@
...
-    class Meta:
-        model = User
-        fields = ['url', 'username', 'email', 'is_staff']
-
-# ViewSets define the view behavior.
-class UserViewSet(viewsets.ModelViewSet):
-    queryset = User.objects.all()
-    serializer_class = UserSerializer
+from rest_framework import routers
+# from ..moviehApp.views import UserViewSet
```
Comment (rimvydascivilis) [inline backend/movieh/urls.py]: remove this

File: backend/movieh/settings.py [hunk 6]
```diff
@@ -11,6 +11,7 @@
 """
 
 from pathlib import Path
+import os
```
Comment (rimvydascivilis) [inline backend/movieh/settings.py:16]: Use https://pypi.org/project/dj-database-url/ for database connection configuration
```suggestion
import dj_database_url
```

Comment (rimvydascivilis) [reply backend/movieh/settings.py:16]: Also install that package and add to `requirements.txt`

File: backend/movieh/settings.py [hunk 7]
```diff
@@ -78,11 +79,11 @@
...
-        'PASSWORD': 'password',
-        'HOST': 'localhost',
-        'PORT': '5432',
+        'NAME': os.getenv('DB_NAME', 'database'),
+        'USER': os.getenv('DB_USER', 'user'),
+        'PASSWORD': os.getenv('DB_PASSWORD', 'password'),
+        'HOST': os.getenv('DB_HOST', 'localhost'),
+        'PORT': os.getenv('DB_PORT', '5432'),
     }
 }
```
Comment (rimvydascivilis) [inline backend/movieh/settings.py]: Its better to have one variable which sets the db connection.
```py
DATABASES = {
    'default': dj_database_url.config(
        default=os.getenv('DATABASE_URL', 'postgres://user:p%23ssword!@localhost/foobar'),
        conn_max_age=600,
        conn_health_checks=True,
    )
}
```

**Review by rimvydascivilis — CHANGES_REQUESTED** [review]

File: backend/movieh/settings.py [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+WSGI_APPLICATION = 'movieh.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/5.1/ref/settings/#databases
+
+DATABASES = {
+    'default': dj_database_url.config(
+        default=os.getenv('DB_CONNECTION', 'postgres://username:password@localhost:port/database'),
```
Comment (rimvydascivilis) [inline backend/movieh/settings.py]: Default url should be valid. Update default url so it connects to local dev db.

**Review by rimvydascivilis — CHANGES_REQUESTED** [review]

File: backend/movieh/settings.py (hunk 8 above)
Comment (PurpJell) [reply backend/movieh/settings.py]: Just to make sure, is this local dev database the one, that i have set up manually or is there another?

Comment (rimvydascivilis) [reply backend/movieh/settings.py]: We can agree to use the following Docker configuration. Update the database URL to `postgresql://user:password@localhost:5432/movieh` and modify `/docker-compose.yml` as follows:
```dockerfile
services:
  frontend:
    build:
      context: ./frontend
      dockerfile: Dockerfile
      args:
        - PORT=80
    networks:
      - app-network
    ports:
      - "80:80"

  database:
    image: postgres:17-alpine
    networks:
      - app-network
    environment:
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
      POSTGRES_DB: movieh
    ports:
      - "5432:5432"
    volumes:
      - db-data:/var/lib/postgresql/data

networks:
  app-network:

volumes:
  db-data:
  ```

**Review by rimvydascivilis — APPROVED** [review]
LTGM

**Pull request merged**